src/main/java/org/jabref/logic/importer/fetcher/DOABFetcher.java

```diff
@@ -0,0 +1,129 @@
+    // private static final String PEER_REVIEW_URL = " https://directory.doabooks.org/rest/peerReviews?";
+
+    @Override
+    public String getName() {
+        return "DOAB";
+    }
+
+    @Override
+    public URL getURLForQuery(QueryNode luceneQuery) throws URISyntaxException, MalformedURLException, FetcherException {
+        URIBuilder builder = new URIBuilder(SEARCH_URL);
+        String query = new DefaultQueryTransformer().transformLuceneQuery(luceneQuery).orElse("");
+        // adding quotations for the query for more specified results
+        // without the quotation the results returned are not relevant to the query
+        query = ("\"".concat(query)).concat("\"");
+        builder.addParameter("query", query);
+        builder.addParameter("expand", "metadata");
+
+        return builder.build().toURL();
+    }
+
+    @Override
+    public Parser getParser() {
+        return InputStream -> {
+            // can't use this method JsonReader.toJsonObject(inputStream) because the results are sent in an array
+            // like format resulting in an error when trying to convert them into a json object
+            // created a similar method suitable for this case "toJsonArray"
+            JSONArray response = JsonReader.toJsonArray(InputStream);
+            if (response.isEmpty()) {
+                return Collections.emptyList();
+            }
+            // single result case
+            if (response.length() == 1) {
+
+                // the information used for bibtex entries are in an array inside the resulting jsonarray
+                // see this query for reference https://directory.doabooks.org/rest/search?query="i open fire"&expand=metadata
+                JSONArray metadataArray = response.getJSONObject(0).getJSONArray("metadata");
+                BibEntry entry = JsonToBibEntry(metadataArray);
+                return Collections.singletonList(entry);
+            }
+            // multiple results
+            List<BibEntry> entries = new ArrayList<>(response.length());
+            for (int i = 0; i < response.length(); i++) {
+                JSONArray metadataArray = response.getJSONObject(i).getJSONArray("metadata");
+                BibEntry entry = JsonToBibEntry(metadataArray);
+                entries.add(entry);
+            }
+            return entries;
+        };
+    }
+
+    private BibEntry JsonToBibEntry(JSONArray metadataArray) {
+        BibEntry entry = new BibEntry();
+        List<Author> authorsList = new ArrayList<>();
+        List<Author> editorsList = new ArrayList<>();
+        StringJoiner keywordJoiner = new StringJoiner(",");
+        for (int i = 0; i < metadataArray.length(); i++) {
+            JSONObject dataObject = metadataArray.getJSONObject(i);
+            switch (dataObject.getString("key")) {
+                case "dc.contributor.author" -> authorsList.add(toAuthor(dataObject.getString("value")));
```

Comment:
In the interest of trying to deepen the comment in https://github.com/JabRef/jabref/pull/8598/files#r839909593

In what circumstance(s) is the `metadataArray` larger than 1 item?

**If** it can't be guaranteed that `metadataArray` is exactly 1 item (in which case we have only one author and one editor fields) I'd probably suggest
`authorsList.addAll(AuthorList.parse(dataObject.getString("value")).getAuthors())` even if it is a bit ugly :/ 

It is probably possible to use StringJoiner with ` and `, but it feels a bit hacky :/ 

I don't know, I guess the best-case scenario is if `metadataArray` is only one item, but I guess we can't always get what we wish for :stuck_out_tongue:

Reply:
Looking at it a bit close, it seems that I've spoken too soon.
@Mohamadi98  do you know if there is any information on what format they are using? 

E.g., 
```
<metadata>
	  <key>dc.contributor.author</key>
	  <language>*</language>
	  <value>Felipe Gonzalez Toro (Ed.)</value>
</metadata>
<metadata>
	<key>dc.contributor.author</key>
	<language>*</language>
	<value>Kamruzzaman, Md. (Liton)</value>
</metadata>
```

Reply:
Also follow-up in https://github.com/JabRef/jabref/issues/8576#issuecomment-1085133569

Reply:
the data itself returned from the API is in xml format, and the actual information that I used for the Bibtex entries are inside an array called metadata which you can add or remove from the results, see this https://directory.doabooks.org/rest/search?query=%22the+deliverance+of+open+access+books%22, so I agree with you it is a little bit weird
I convert the format to json, It's easier for me to work with json

Reply:
@Mohamadi98  Most entries seem to be already  in correct format, however some seem to indicate Ediors in the authors field
We suggest you apply a minimal heuristic:
1. Check if the dc.contirbutor.author contains (Ed.)  
2. if yes -> remove that (Ed.) part 
3. Parse the value now using Authorlist.parse() it takes care of different names formats 
4. Repeat for other editors
6. Put in the editors field

The other normal case:
1. Check if the dc.contirbutor.author contains (Ed.)  
2. If no ->  Parse it using AuthorLIst.parse() 
4. Repeat for other authors
5. Put in the authors field

It's a bit of a mess with a list of list of authors... and the naming doesn't really help (yep, it's a mess^^)

But if you're stuck, just ask @k3KAW8Pnf7mkmdSMPHz27

Reply:
The AuthorList.parse can also handle names not in (LastName, FirstName) format you will see in one of the tests in the latest commit that there are names that returns from the API in (FirstName LastName) and are parsed correctly, so this nice